runtime-decompiler/src/main/java/org/jrd/backend/communication/FsAgent.java

```diff
@@ -182,6 +181,22 @@ private T onEntryList(ZipInputStream zipInputStream, String clazz, CpOperator<T>
             }
             return null;
         }
+
+        private T onEntryOther(File f, String clazz, CpOperator<T> op) throws IOException {
```

Comment:
Can this operate on any file or just ZipFile? If just zip, then maybe the method signature should change

Reply:
This is the old method extracted into function. Other functions called from onEntryOther require the archive file, thus the use of File.

Reply:
Even if it is 
 private T onEntryOther(ZIpFile f, String clazz, CpOperator<T> op) 
it still can be called as
  T ret = onEntryOther(new ZipFile(f);, clazz, op);
I do not have strong preffernce. Maybe  best woudl be
```
 private T onEntryOther(File f, String clazz, CpOperator<T> op)  {
    return onEntryOther(new ZipFile(f);, clazz, op)
}
 private T onEntryOther(ZIpFile f, String clazz, CpOperator<T> op)  {
....
```

Although it is severe nitpicking, it somehow gave sense

Reply:
Because op.onJarEntry requires the File, not the ZipFile when op is instance of WriteingCpOperator, I have to pass the file into the function. op.onJarEntry takes File, ZipFile, ZipEntry and String. Even though I don't see the String being used in any of CpOperator implementations.

Reply:
please do the cosmetic change I suggested above

Reply:
I am not sure if we understand each other. The onEntryOther needs the File variable, as it is used as a argument in another function call that onEntryOther is doing - line 194. I can't omit the File from here.

Reply:
You mean ```                         return op.onJarEntry(f, zipFile, ze, clazz);``` or something more? Originally I had in mind to elaborate also on this. But nvm. Drop this request